stdlib/public/core/CompilerProtocols.swift

```diff
@@ -396,9 +396,12 @@ public protocol ExpressibleByExtendedGraphemeClusterLiteral
 }
 
 extension ExpressibleByExtendedGraphemeClusterLiteral
-  where Self.ExtendedGraphemeClusterLiteralType == Character {
-  public init(unicodeScalarLiteral value: UnicodeScalar) {
-    self.init(extendedGraphemeClusterLiteral: Character(value))
+where ExtendedGraphemeClusterLiteralType == UnicodeScalarLiteralType {
+
+  @effects(readonly)
```

Comment:
Please don't specify this. You can't guarantee the initializer you chain to follows it.